src/main/java/com/cleanroommc/groovyscript/compat/mods/essentialcraft/MithrilineFurnace.java

```diff
@@ -0,0 +1,106 @@
+package com.cleanroommc.groovyscript.compat.mods.essentialcraft;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import essentialcraft.api.MithrilineFurnaceRecipe;
+import essentialcraft.api.MithrilineFurnaceRecipes;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import org.jetbrains.annotations.Nullable;
+
+@RegistryDescription
+public class MithrilineFurnace extends VirtualizedRegistry<MithrilineFurnaceRecipe> {
+
+    @RecipeBuilderDescription(example = @Example(".input(item('minecraft:coal_block') * 3).output(item('minecraft:diamond_block')).espe(500)"))
+    public MithrilineFurnace.RecipeBuilder recipeBuilder() {
+        return new MithrilineFurnace.RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(MithrilineFurnaceRecipes::removeRecipe);
+        restoreFromBackup().forEach(MithrilineFurnaceRecipes::addRecipe);
+    }
+
+    @MethodDescription(example = @Example("ore('dustGlowstone')"))
+    public boolean removeByInput(IIngredient x) {
+        return MithrilineFurnaceRecipes.RECIPES.removeIf(r -> {
+            if (x.equals(r.input)) {

```

Comment:
this doesnt work and it always false because `IIngredient` is at no point equal to an `Ingredient`. have to do
`for (var item : r.input.getMatchingStacks()) { x.test(item) ...` instead.

Reply:
Uh oh, how did it work when I tested it....